src/HeaderCell.ts

```diff
@@ -0,0 +1,60 @@
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import { RegistryMixin } from '@dojo/widget-core/mixins/Registry';
+import { v } from '@dojo/widget-core/d';
+import { HasColumn, HasSortDetail, HasSortEvent } from './interfaces';
+import { ThemeableMixin, theme, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+
+import * as headerCellClasses from './styles/headerCell.css';
+
+export interface HeaderCellProperties extends ThemeableProperties, HasColumn, HasSortDetail, HasSortEvent { }
+
+export const HeaderCellBase = ThemeableMixin(RegistryMixin(WidgetBase));
+
+@theme(headerCellClasses)
+class HeaderCell extends HeaderCellBase<HeaderCellProperties> {
+	onSortRequest(): void {
+		const {
+			key = '',
+			sortDetail,
+			onSortRequest
+		} = this.properties;
+
+		onSortRequest({
+			columnId: key,
+			descending: Boolean(sortDetail && sortDetail.columnId === key && !sortDetail.descending)
+		});
+	}
+
+	render() {
+		const {
+			key,
+			column,
+			sortDetail,
+			onSortRequest
+		} = this.properties;
+
+		const classes = [ headerCellClasses.headerCell, column.sortable !== false ? headerCellClasses.sortable : null ];
+
+		const sortClasses = [
+			sortDetail ? headerCellClasses.sortArrow : null,
+			sortDetail && sortDetail.descending ? headerCellClasses.sortArrowDown : null,
```

Comment:
Since these all depend on sortDetail, I did a ternary that either creates this array or an empty array, then used your ternary approach. It does read much better. Thanks.